### PR TITLE
feat: Add sidetree client

### LINKED ISSUE #36: Add sidetree client

### PR DESCRIPTION
**Pull request by fqutishat**
closes #36

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: component/sidetree/client.go [hunk 1]
```diff
@@ -0,0 +1,638 @@
...
+	}
+
+	didSuffix, err := getUniqueSuffix(did)
+	if err != nil {
+		return nil, err
+	}
+
+	revealValue := updateDIDOpts.RevealValue
+
+	// TODO: client should be managing reveal value, this defaulting here is just temporary convenience
```
Comment (sandrask) [inline component/sidetree/client.go:361]: Please create ticket to remove this code and force the clients of this sidetree client to manage reveal value. It is referenced in update, deactivate and recover.

**Review by sandrask — APPROVED** [review]

File: component/sidetree/doc/doc.go [hunk 2]
```diff
@@ -0,0 +1,158 @@
+/*
```
Comment (troyronda) [inline component/sidetree/doc/doc.go:1]: sidetree should not be at the root component folder.

File: component/sidetree/option/create/option.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+// WithService add service.
+func WithService(service *docdid.Service) Option {
+	return func(opts *Opts) {
+		opts.Services = append(opts.Services, *service)
+	}
+}
+
+// WithSidetreeEndpoint sidetree url.
+func WithSidetreeEndpoint(sidetreeEndpoint string) Option {
```
Comment (troyronda) [inline component/sidetree/option/create/option.go:48]: We should also have a way to inject an endpoint discovery mechanism.

Comment (fqutishat) [reply component/sidetree/option/create/option.go]: done

File: component/sidetree/doc/doc.go (hunk 2 above)
Comment (fqutishat) [reply component/sidetree/doc/doc.go:1]: @troyronda what is the better location for generic sidetree client

File: component/sidetree/client.go (hunk 1 above)
Comment (fqutishat) [reply component/sidetree/client.go]: @sandrask I force clients of this sidetree client to manage reveal value to manage reveal value

File: component/sidetree/doc/doc.go (hunk 2 above)
Comment (fqutishat) [reply component/sidetree/doc/doc.go]: @troyronda Done. I move it under vdr package

**Review by troyronda — APPROVED** [review]

**Review by sandrask — APPROVED** [review]

**Pull request merged**